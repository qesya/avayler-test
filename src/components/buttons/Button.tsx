import React from 'react'
import clsx from 'clsx';

interface IButton {
  title: string;
  variant: 'outline' | 'solid';
  onClick: () => void;
}

const Button: React.FC<IButton> = ({ title, onClick, variant }) => {
  return (
    <div>
      <button
        data-testid="Test-Button"
        className={
          clsx({
            "border-white border-2 py-4 px-16 flex flex-row md:self-start mt-10 rounded-md w-full hover:bg-white hover:text-black transition duration-500 ease-in-out" : variant === "outline",
            "bg-red-600 py-4 px-16 flex self-start mt-10 rounded-md": variant === "solid"
          })
        }
        onClick={onClick}>
        <p className="font-AbrilFatface text-white hover:text-black w-full" data-testid="Test-Title">{title}</p>
      </button>
    </div>
  )
}

export default Button
