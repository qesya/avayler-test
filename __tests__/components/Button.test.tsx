import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Button from '../../src/components/buttons/Button';

describe('Component:Button', () => {
  it('renders without crashing', async () => {
    const component = render(<Button variant="solid" onClick={() => { }} title="test" />);
    expect(component).toMatchSnapshot();
  });

  it('it should render correct button label', async () => {
    const component = render(<Button variant="solid" onClick={() => { }} title="test" />);
    const expected = "test"
    expect(component.getByTestId('Test-Title').textContent).toContain(expected);
  });

  it('it should click button', async () => {
    const expected = jest.fn();
    const { getByTestId } = render(<Button variant="solid" onClick={expected} title="test" />);
    fireEvent.click(getByTestId('Test-Button'));
    expect(expected).toHaveBeenCalled();
  });
});