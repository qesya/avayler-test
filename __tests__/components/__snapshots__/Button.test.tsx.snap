// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:Button renders without crashing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <button
          class="bg-red-600 py-4 px-16 flex self-start mt-10 rounded-md"
          data-testid="Test-Button"
        >
          <p
            class="font-AbrilFatface text-white hover:text-black w-full"
            data-testid="Test-Title"
          >
            test
          </p>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <button
        class="bg-red-600 py-4 px-16 flex self-start mt-10 rounded-md"
        data-testid="Test-Button"
      >
        <p
          class="font-AbrilFatface text-white hover:text-black w-full"
          data-testid="Test-Title"
        >
          test
        </p>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
